sharding/collator.go

```diff
@@ -82,6 +93,25 @@ func checkShardsForProposal(c collatorClient, head *types.Header) error {
 	return nil
 }
 
+// checkForValidators checks if the client is in the validator pool because
+// we can't guarantee our tx for deposit will be in the next block header we receive
+// The function calls IsValidatorDeposited from the VMC and returns true if
+// the client is in the validator pool
+func checkForValidators(c collatorClient) (bool, error) {
```

Comment:
Please make this method name more descriptive. Maybe isAccountInValidatorSet or something along those lines?